### PR TITLE
fix(hatchery/kubernetes): delete secret

### PR DESCRIPTION
**Pull request by yesnault**

### CODE REVIEW

File: engine/hatchery/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -332,9 +331,8 @@ func (h *HatcheryKubernetes) SpawnWorker(ctx context.Context, spawnArgs hatchery
 			Namespace:                  h.Config.Namespace,
 			DeletionGracePeriodSeconds: &gracePeriodSecs,
 			Labels: map[string]string{
-				LABEL_HATCHERY_NAME:     h.Configuration().Name,
-				LABEL_WORKER_NAME:       workerConfig.Name,
-				LABEL_WORKER_MODEL_PATH: slug.Convert(spawnArgs.Model.GetPath()),
```
Comment (fsamin) [inline engine/hatchery/kubernetes/kubernetes.go]: keep LABEL_WORKER_MODEL_PATH

**Review by sguiheux — APPROVED** [review]

File: engine/hatchery/kubernetes/secrets.go [hunk 2]
```diff
@@ -59,11 +61,31 @@ func (h *HatcheryKubernetes) deleteSecrets(ctx context.Context) error {
...
+				return sdk.WrapError(err, "cannot delete secret %s from worker %s", secret.Name, secretLabels[LABEL_WORKER_NAME])
+			}
+			// no break, we can have two secrets used by a worker (registry and config secrets)
+		}
+	}
 
+	return nil
+}
+
+func (h *HatcheryKubernetes) createRegistrySecret(ctx context.Context, model sdk.WorkerStarterWorkerModel, workerConfig workerruntime.WorkerConfig) (string, error) {
```
Comment (fsamin) [inline engine/hatchery/kubernetes/secrets.go:87]: replace workerConfig object by just `workerName string`

File: engine/hatchery/kubernetes/kubernetes_test.go [hunk 3]
```diff
@@ -96,7 +96,6 @@ func TestHatcheryKubernetes_Status(t *testing.T) {
 			require.Equal(t, "666", podRequest.Labels[hatchery.LabelServiceJobID])
 			require.Equal(t, "999", podRequest.Labels[hatchery.LabelServiceNodeRunID])
 			require.Equal(t, "my-worker", podRequest.Labels[LABEL_WORKER_NAME])
-			require.Equal(t, "group-model1", podRequest.Labels[LABEL_WORKER_MODEL_PATH])
```
Comment (fsamin) [inline engine/hatchery/kubernetes/kubernetes_test.go:99]: +

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19268.0 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build SMTPMock ■   
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#19268.0 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build SMTPMock ■   
  * Lint UI Translate ✔ 
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

File: engine/hatchery/kubernetes/secrets.go (hunk 2 above)
Comment (yesnault) [reply engine/hatchery/kubernetes/secrets.go]: done

File: engine/hatchery/kubernetes/kubernetes.go (hunk 1 above)
Comment (yesnault) [reply engine/hatchery/kubernetes/kubernetes.go]: done

File: engine/hatchery/kubernetes/kubernetes_test.go (hunk 3 above)
Comment (yesnault) [reply engine/hatchery/kubernetes/kubernetes_test.go]: done

**Comment by yesnault** [comment]
close #6653

**Review by richardlt — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Pull request merged**